strawberry/printer.py

```diff
@@ -35,33 +40,77 @@
     from strawberry.schema import BaseSchema
 
 
-def print_schema_directive_params(params: Dict) -> str:
+@dataclasses.dataclass
+class PrintExtras:
+    directives: Set[str] = dataclasses.field(default_factory=set)
+    types: Set[type] = dataclasses.field(default_factory=set)
+
+
+def _serialize_dataclasses(value: Any) -> Any:
```

Comment:
Can we narrow the type definitions here?
```suggestion
def _serialize_dataclasses(value: object) -> Union[dict, list]:
```

Reply:
The problem is that value can actually be anything (i.e. `str`, `int`, an object that is a dataclass, `list`, `dict`)

But I can improve it using overloads, which will match the actual typing closer

Reply:
Right, that's why I think we want `object`. It ensures that anything done to `value` within the function has to be possible on _all_ `object`s. If you use `Any`, the type checker just gives up and lets you do whatever you want

Reply:
It's done!